src/Network/Consul.hs

```diff
@@ -37,86 +44,326 @@ import Control.Monad (forever)
 import Control.Monad.IO.Class
 import Control.Monad.Catch (MonadMask)
 import Control.Retry
+import Data.Aeson (Value(..), decode,encode)
+import qualified Data.ByteString as B
+import qualified Data.ByteString.Lazy as BL
+import qualified Data.HashMap.Strict as H
+import Data.Maybe (catMaybes, isJust, listToMaybe)
 import Data.Monoid ((<>))
 import Data.Text (Text)
 import qualified Data.Text as T
 import qualified Data.Text.Read as TR
 import Data.Word
+import qualified Data.Vector as V
 import qualified Network.Consul.Internal as I
 import Network.Consul.Types
-import Network.HTTP.Client (Manager)
+import Network.HTTP.Client -- (method, Manager, responseBody)
 import Network.HTTP.Client.TLS (newTlsManager, newTlsManagerWith, tlsManagerSettings)
+import Network.HTTP.Types
 import Network.Socket (PortNumber)
 import UnliftIO (MonadUnliftIO, async, cancel, finally, wait, waitAnyCancel, withAsync)
 
+import Network.Consul.Internal
 
 import Prelude hiding (mapM)
 
 parseTtl :: Integral t => Text -> t
 parseTtl ttl = let Right (x,_) = TR.decimal $ T.filter (/= 's') ttl in x
 
+-- QUESTION: rename `man` to `mManager` or `maybeMan` or `maybeManager` or `mm` or `mman` or `mmanager`?
 initializeConsulClient :: MonadIO m => Text -> PortNumber -> Maybe Manager -> m ConsulClient
 initializeConsulClient hostname port man = do
   manager <- liftIO $ case man of
                         Just x -> return x
                         Nothing -> newTlsManager
   return $ ConsulClient manager hostname port False
+                                         --   ^--- does this mean the client implemented here
+                                         --        does not support TLS?
```

Comment:
Yes, and it's `True` in `initializeTlsConsulClient` just below.

Personally, in all programming languages, I don't like constructing things from positional "literal" arguments because reading the code you have no idea what that `True` is for. E.g. `shootTheRockets True False`, who knows what they do.

I'd prefer either

```haskell
let withTls = False
return $ ConsulClient manager hostname port withTls
```

to give the concept a name, or to directly use a record construction like

```haskell
return $
  ConsulClient
    { ccManager = manager
    , ccHostname = hostname
    , ccPort = port
    , ccWithTls = False
    }
```

Reply:
Great example, thanks!